File: Easy/Temperatures/C#/Temperatures.cs
using System;

class Solution
{
    static void Main(string[] args)
    {
        int countOfTemperatures = int.Parse(Console.ReadLine());
        if (countOfTemperatures == 0)
            Console.WriteLine("0");
        else
        {
            int[] temps = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);
            
            int nearestToZero = int.MaxValue;
            for (int i = 0; i < temps.Length; i++)
                nearestToZero = 
                Math.Abs(temps[i]) < Math.Abs(nearestToZero) ? temps[i] :
                Math.Abs(temps[i]) > Math.Abs(nearestToZero) ? nearestToZero : 
                Math.Max(temps[i], nearestToZero);
            
            Console.WriteLine(nearestToZero);
        }
    }
}

File: Easy/Temperatures/Python 3/Temperatures.py
import sys
import math

count_of_temperatures = int(input())
if (count_of_temperatures == 0):
    print("0");
else:
    temps = input().split(" ")  # the n temperatures expressed as integers ranging from -273 to 5526
    
    nearest_to_zero = 5527;
    for i in range(count_of_temperatures):
        if abs(int(temps[i])) < abs(nearest_to_zero):
            nearest_to_zero = int(temps[i]);
        elif abs(int(temps[i])) == abs(nearest_to_zero):
            nearest_to_zero = max(int(temps[i]), nearest_to_zero);
    
    print(nearest_to_zero)